numba/utils.py

```diff
@@ -561,6 +560,7 @@ def __init__(self, obj, func, *args, **kwargs):
             import atexit
             atexit.register(self._exitfunc)
             finalize._registered_with_atexit = True
+            atexit.register(_at_shutdown)
```

Comment:
Then I would suggest creating a dummy finalizer to ensure the callback is always registered. Otherwise using `shutting_down()` will have surprising effects.